### PR TITLE
Better naming and values for MockEnv parameters + unit tests

### PR DESCRIPTION
**Pull request by motus**
* Use `mock_env_range`, `mock_env_seed`, and `mock_env_metrics` instead of just `range`, `seed`, and `metrics` to avoid naming conflicts.
* Use integer `mocl_env_seed` values to distinguish between deterministic behavior, random seed, and user-specified seed for `MockEnv`.
* Update unit tests and configs to validate the seed behavior.
* Add new unit tests for different initializations of the tunable values

### CODE REVIEW

File: mlos_bench/mlos_bench/tests/config/tunable-values/tunable-values-example.jsonc [hunk 1]
```diff
@@ -2,8 +2,8 @@
 {
     "$schema": "https://raw.githubusercontent.com/microsoft/MLOS/main/mlos_bench/mlos_bench/config/schemas/tunables/tunable-values-schema.json",
 
-    "sched_migration_cost_ns": 500000,
-    "sched_latency_ns": 12000000,
+    "sched_migration_cost_ns": 400000,
+    "sched_latency_ns": 10000000,
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/tests/config/tunable-values/tunable-values-example.jsonc:6]: Are these changing to ensure they?

File: mlos_bench/mlos_bench/tests/config/cli/mock-bench.jsonc [hunk 2]
```diff
@@ -14,9 +14,10 @@
 
     "environment": "environments/mock/mock_env.jsonc",
 
-    "tunable_values": [
-        "tunable-values/tunable-values-example.jsonc"
-    ],
+    // Use default values for tunables unless otherwise specified
+    // "tunable_values": [
+    //     "tunable-values/tunable-values-example.jsonc"
+    // ],
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/tests/config/cli/mock-bench.jsonc:20]: Let's add a Launcher test that verifies that the implicit Optimizer and TunableGroups were initialized with the defaults.

Comment (bpkroth) [reply mlos_bench/mlos_bench/tests/config/cli/mock-bench.jsonc:20]: Or at least double check that there is one for that already.

File: mlos_bench/mlos_bench/tests/config/tunable-values/tunable-values-example.jsonc (hunk 1 above)
Comment (motus) [reply mlos_bench/mlos_bench/tests/config/tunable-values/tunable-values-example.jsonc:6]: ...to ensure they are different from the default values that we have in `tunables.json`

File: mlos_bench/mlos_bench/tests/config/cli/mock-bench.jsonc (hunk 2 above)
Comment (motus) [reply mlos_bench/mlos_bench/tests/config/cli/mock-bench.jsonc:20]: yep, that was the whole point. You'll see the new unit tests below.

Comment (bpkroth) [reply mlos_bench/mlos_bench/tests/config/cli/mock-bench.jsonc:20]: Ah, yep, I see it now.  Thanks!

File: mlos_bench/mlos_bench/tests/config/tunable-values/tunable-values-example.jsonc (hunk 1 above)
Comment (bpkroth) [reply mlos_bench/mlos_bench/tests/config/tunable-values/tunable-values-example.jsonc:6]: Sorry, seems my comment got lost.  Yes, that's what I was asking for.  Can you just add a comment to that effect so we know why.

File: mlos_bench/mlos_bench/tests/config/tunable-values/tunable-values-example.jsonc [hunk 3]
```diff
@@ -2,8 +2,8 @@
 {
     "$schema": "https://raw.githubusercontent.com/microsoft/MLOS/main/mlos_bench/mlos_bench/config/schemas/tunables/tunable-values-schema.json",
 
-    "sched_migration_cost_ns": 500000,
-    "sched_latency_ns": 12000000,
+    "sched_migration_cost_ns": 400000,
```
Comment (bpkroth) [inline mlos_bench/mlos_bench/tests/config/tunable-values/tunable-values-example.jsonc:5]:
```suggestion
    // Values that are different from the defaults
    // in order to test --tunable-values handling in OneShotOptimizer.
    "sched_migration_cost_ns": 400000,
```

File: mlos_bench/mlos_bench/tests/config/tunable-values/tunable-values-example.jsonc (hunk 1 above)
Comment (bpkroth) [reply mlos_bench/mlos_bench/tests/config/tunable-values/tunable-values-example.jsonc:6]: I added the comment in the suggestion above.

**Review by bpkroth — APPROVED** [review]

**Pull request merged**